src/authorization.js

```diff
@@ -525,13 +525,14 @@ class Authorization {
   }
 
   /**
-   * Sets the agent WebID for this authorization. Implemented as `setAgent()`
-   * setter method to enforce mutual exclusivity with `group` property, until
-   * ES6 setter methods become available.
+   * Sets the agent WebID for this authorization.
    * @method setAgent
-   * @param agent {String|Quad} Agent URL (or `acl:agent` RDF triple).
+   * @param agent {string|Quad|GroupListing} Agent URL (or `acl:agent` RDF triple).
    */
   setAgent (agent) {
```

Comment:
I'm concerned that these `setAgent`/`setGroup` methods are adding a fair bit of complexity for not much payoff. Why should permissions be mutable? Can't we handle the modification of an ACL at the `PermissionSet` level?